etna/records/tests/test_templatetags.py

```diff
@@ -267,3 +623,54 @@ def test_non_reference_number_url(self):
                     record_url(source.repository, use_non_reference_number_url=True),
                     expected_result,
                 )
+
+    def test_is_page_current_item_in_hierarchy(self):
+        page = self.tna_long_hierarchy_record_instance
+        hierarchy = page.hierarchy
+        for name, level_item, expected_result in (
+            ("item_level_1", hierarchy[0], False),
+            ("item_level_3", hierarchy[1], False),
+            ("item_level_6", hierarchy[2], False),
+            ("item_level_7", hierarchy[3], True),
+        ):
+            with self.subTest(name):
+                self.assertEqual(
+                    is_page_current_item_in_hierarchy(page, level_item), expected_result
+                )
+
+    def test_level_name(self):
+        for current_record, expected_result in (
+            (self.tna_record_instance, "Series"),
+            (self.record_instance, "Sub-fonds"),
+        ):
+            with self.subTest(self):
+                # We pass in the current record level code and is_tna value
+                # and this function then retrieves the level name associated
+                # with that level code/record
+                self.assertEqual(
+                    level_name(current_record.level_code, current_record.is_tna),
+                    expected_result,
+                )
+
+    def test_breadcrumb_items(self):
+        for current_record, expected_result in (
+            (self.tna_long_hierarchy_record_instance, ["C162", "C9685", "C8077549"]),
+            (
+                self.long_hierarchy_record_instance,
+                [
+                    "16131aa0-f1d9-42a5-8488-e9a236366b4b",
+                    "2dba8c17-0f69-4a53-b918-e6ddb06c41a7",
+                    "66787951-2237-4f8a-882f-0ac275fe2bff",
+                ],
+            ),
+        ):
+            with self.subTest(self):
+                # We pass in the record hierarchy, is_tna value and the current record
+                # and this function then retrieves the breadcrumb items associated
+                # with that hierarchy, depending on the state of is_tna
+                for i, item in enumerate(
+                    breadcrumb_items(
+                        current_record.hierarchy, current_record.is_tna, current_record
+                    )
+                ):
+                    self.assertEqual(item.iaid, expected_result[i])
```

Comment:
```suggestion
    def test_breadcrumb_items(self):
record = self.tna_long_hierarchy_record_instance
        hierarchy = page.hierarchy
        item_level_1 = hierarchy[0]
        item_level_3 = hierarchy[1]
        expected_breadcrumb_items = [item_level_1, item_level_3, record]
        self.assertEqual(breadcrumb_items(hierarchy, page.is_tna, record), expected_breadcrumb_items)
```
@jamesbiggs , nice to see this test case.
Similar to is_page_current_item_in_hierarchy test, lets avoid looping `for i, item in enumerate( breadcrumb_items(..` 
to call breadcrumb_items as breadcrumb_items is called only once

See suggested changes for that

Reply:
In that case, I did for 1 bit nonTna, I  would suggest add TNA/NonTNA in the similar suggested way - record_tna, record_non_tna

Reply:
Done 😃